ops/ccs-ops-misc/load_test/common/config.py

```diff
@@ -1,73 +1,85 @@
-import os
+'''Load and save configuration file.
+'''
+
 import yaml
 
-'''
-Saves a config file using the input file data.
-'''
-def save(fileData):
-    configFile = open('config.yml', 'w')
-    configFile.write("homePath: \"%s\"\n" % fileData["homePath"])
-    configFile.write("clientCertPath: \"%s\"\n" % fileData["clientCertPath"])
-    configFile.write("serverPublicKey: \"%s\"\n" % fileData["serverPublicKey"])
-    configFile.write("dbUri: \"%s\"\n" % fileData["dbUri"])
-    configFile.write("testHost: \"%s\"\n" % fileData["testHost"])
-    configFile.write("testRunTime: \"%s\"\n" % fileData["testRunTime"])
-    configFile.write("testNumTotalClients: \"%s\"\n" % fileData["testNumTotalClients"])
-    configFile.write("testCreatedClientsPerSecond: \"%s\"\n" % fileData["testCreatedClientsPerSecond"])
-    configFile.write("resetStatsAfterClientSpawn: \"%s\"" % fileData["resetStatsAfterClientSpawn"])
-    configFile.close()
+def save(file_data):
```

Comment:
I realize this is an improvement over what already existed, but I think this function could be made a _little_ safer without much work:
1. I think the zeitgeist (_or_ [PEP-343](https://peps.python.org/pep-0343)) would have you wrap the file i/o `open(...)` in a `with` block, e.g.
```python
with open('config.yml', 'w', encoding='utf-8') as config:
    # do stuff with the scoped `config` variable, but worry not about
    # having to explicitly issue `.close()`
```
Otherwise, I think you might consider wrapping it in a try/except/finally or simply try/finally to always close the file.

2. I don't know if I've ever really been _in_ the python game, but last time I checked, `ruamel` seemed to be a feature-complete for yaml marshalling/unmarshalling. I would think that `PyYAML==6.0` is also capable. I'd imagine something like the following would be more auspicious than manually coercing some string into a yaml config file, e.g.

```python
def save(file_data): # assuming file_data is indeed a dict
    with open('config.yml', 'w') as config:
        yaml.dump(file_data, config, default_flow_style=False)
```

Reply:
You're exactly right. We're already using PyYAML in the load function, so I went with that.